### ISSUE TITLE
Model support: GPT

### ISSUE THREAD

**Issue by bonham79** [CUNY-CL/yoyodyne#159]
MIght as well set up an autoregressive decoder since T5 is on the docket. This shouldn't be too much of a hassle since the Transformer model works, but leaving as an open issue to do validation testing on.

**Comment by kylebgorman** [comment]
Dumb question but how is this different than the type of decoder-only LM we were talking about?

**Comment by bonham79** [comment]
It's exactly that. It's just running transformer with --encoder_layers=0. Why I'm saying it shouldn't be much of a hassle (technically done already, just needs some benchmarking).

**Comment by kylebgorman** [comment]
I think Adam has an implementation in his fork, but hasn’t PRed it yet.

**Comment by Adamits** [comment]
Yes, though I specifically have a prefix-LM. This can be used like GPT if you just ensure the prefix is always 0. I have some currently dirt code that takes a source and target, and concatenates them and always assumes the source is the prefix for training.

I can work on a PR in the next few weeks.

**Comment by bonham79** [comment]
Perfect, give me a ping when ready and i'll do some benchmarking at home. 

Any issue in adding features to the prefix concat? Should allow an easy hack of doing task/multitask specific training (just make a treat a target task as a feature in training).

**Comment by Adamits** [comment]
Sorry, yes, the features are in the prefix too by default.
